flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -366,11 +366,10 @@ private void registerAsyncCheckpointRunnable(long checkpointId, AsyncCheckpointR
 		synchronized (lock) {
 			if (closed) {
 				LOG.debug("Cannot register Closeable, this subtaskCheckpointCoordinator is already closed. Closing argument.");
-				final boolean running = asyncCheckpointRunnable.isRunning();
 				closeQuietly(asyncCheckpointRunnable);
 				checkState(
-					!running,
-					"SubtaskCheckpointCoordinatorImpl was closed without closing asyncCheckpointRunnable %s",
+					!checkpoints.containsKey(checkpointId),
+					"SubtaskCheckpointCoordinator was closed without releasing asyncCheckpointRunnable, %s",
```

Comment:
nit: I guess it seem not very readable for message like `.....releasing asyncCheckpointRunnable, 12`. Maybe change to `......releasing asyncCheckpointRunnable for checkpoint 12`? (Take checkpointId = 12 as example).